react-js/react.md

```diff
@@ -41,29 +42,108 @@ You will spend the day working through these materials.
 
 ### Topics to learn & Facilitate
 
-- What is React?
-    - React is a library for creating User Interfaces.
-    - This means it is used for creating what some refer to as the "view layer" of an application.
-- JSX adds to our JavaScript to allow for HTML-like markup in JavaScript.
-- Show Hello World example, plus expanded example.
-- Example for appending to a list in jQuery vs. the same in React.
+- React lets you generate HTML using JavaScript
+- The HTML can change over time, unlike regular HTML
+- No jQuery needed
+
+### Examples
+- Hello World component
+```
+helloWorld.js
+
+import ReactDOM from 'react-dom'
+
+const HelloWorld = () => <h1>Hello, World!</h1>
+      
+ReactDOM.render(<HelloWorld />, document.getElementById('app'))
+
+index.html (The rendering of the above React code will be done in this file)
+
+<div id = "app"></div>
+
+```
+
+- Component with a single prop
+``` 
+welcome.js
+
+import ReactDOM from 'react-dom'
+
+function Welcome(props) {  
+return <h1>Hello, {props.name}</h1>;
+}
+
+const element = <Welcome name="ABC" />;
+ReactDOM.render(
+  element,
+  document.getElementById('root')
+);
+
+index.html (The rendering of the above React code will be done in this file)
 
+<div id = "root"> </div>
+
+```
+
+- Component assigning a `className`
+```
+// ParentComponent.js
+
+import React from 'react';
+import { ChildComponent } from '../child/ChildComponent';
+
+export const ParentComponent = () =>
+  <div className="parent-component">
+    <ChildComponent className="parent-component__child">
+      ...
+    </ChildComponent>
+  </div>
+
+// ChildComponent.js
+
+import React from 'react';
+
+export const ChildComponent = ({ className, children }) =>
+  <div className={`some-css-className ${className}`}>
+    {children}
+  </div>
+  
+```
+will render as:
+```
+<div class="parent-component">
+  <div class="some-css-className parent-component__child">
+    ...
+  </div>
+</div>
+
+```
 
 ### Common Mistakes / Misconceptions
 
-1. JSX tag names for components must match up with component names, both in spelling and case.
-2. In render functions, don't forget to return JSX.
-3. Don't forget to specify both arguments in ReactDOM.render function.
-4. Don't forget `this` when working within a React class. Examples: `this.state.thing`
-5. Don't forget that variables in JSX must be enclosed in brackets and don't forget number 4. Example: `<div>Hello {this.state.thing}</div>`
-6. The event system in React uses attributes you are likely familiar with from HTML, but they are written using camelCase. Example: `<button onClick={function() { alert('hello!') }}>I am a button</button>`
-7. The term 'ReactJS' is commonly used to clarify that it's a javascript framework; however, 'React' is its official name.
+1. In render functions, don't forget to return JSX.
```

Comment:
Please mention the error they will see and suggest they try it out intentionally. Mention you can return null intentionally for an empty output

Reply:
Okay